sfdoc/publish/html.py

```diff
@@ -94,35 +94,44 @@ def same(item1, item2):
                 return True
             else:
                 return item1 == item2
-        return same(
+        rc = same(
             self.author,
             record[settings.SALESFORCE_ARTICLE_AUTHOR_FIELD],
-        ) and same(
+        )
+        rc &= same(
             self.author_override,
             record[settings.SALESFORCE_ARTICLE_AUTHOR_OVERRIDE_FIELD],
-        ) and same(
+        )
+        rc &= same(
             self.is_visible_in_csp,
             record['IsVisibleInCsp'],
-        ) and same(
+        )
+        rc &= same(
             self.is_visible_in_pkb,
             record['IsVisibleInPkb'],
-        ) and same(
+        )
+        rc &= same(
             self.is_visible_in_prm,
             record['IsVisibleInPrm'],
-        ) and same(
+        )
+        rc &= same(
             self.title,
             record['Title'],
-        ) and same(
+        )
+        rc &= same(
             self.summary,
             record['Summary'],
-        ) and same(
+        )
+        rc &= same(
```

Comment:
This is relying on the esoteric fact that Python booleans can act as integers, and I don't think it's an improvement in readability.

Reply:
This particular change was more about line-by-line debuggabiity than readability but sure, I can switch to the rc = rc and same(...) syntax.

If you programmed in Python before Pep 285 type, the fact that booleans are essentially just integers might not seem esoteric! ;)

```suggestion
        rc = rc and same(
```

You inspired me to do a bit of research though.

For the record though, a&=b means almost the same thing in Javascript which had booleans from the beginning. It does end up coercing to int rather than overloading boolean. I suspect that a&=b "works" for booleans in any language that has a C-inspired & operator. 

The ampersand operator is actually overloaded and not just a side-effect of the underlying implementation.

https://github.com/python/cpython/blob/530f506ac91338b55cf2be71b1cdf50cb077512f/Objects/boolobject.c

But either way, its a rarely used operator so I can see why it would seem esoteric. I can change it.